tests/ignite/metrics/test_confusion_matrix.py

```diff
@@ -647,6 +654,132 @@ def _test_distrib_accumulator_device(device):
         ), f"{type(cm.confusion_matrix.device)}:{cm._num_correct.device} vs {type(metric_device)}:{metric_device}"
 
 
+def test_multiLabelConfusionMatrix():
+
+    output = torch.Tensor([[1, 1, 0], [0, 1, 0], [1, 0, 1], [1, 1, 1]])
+    target = torch.Tensor([[0, 1, 0], [0, 0, 1], [1, 0, 1], [1, 0, 1]])
+
+    mtr = MultiLabelConfusionMatrix(3, normalized=False)
+    mtr.update([output, target])
+
+    conf_mtrx = mtr.compute()
+    correct_conf_mtrx = torch.tensor([[[1, 1], [0, 2]], [[1, 2], [0, 1]], [[1, 0], [1, 2]]])
+
+    TestCase.assertTrue(
```

Comment:
in general we use pytest and do simply `assert something, err message`, so let's in the same way here.

About the tests, I'm thinking if it won't be interesting to put them into a separate file : `test_multilabel_confusion_matrix.py`. Maybe, same for the code of `MultiLabelConfusionMatrix`.

In the tests, you can simply generate inputs, compute the result and compare it vs sklearn result as it is done for ConfusionMatrix.
Do you think that it would make sense to ensure that MultiLabelConfusionMatrix works and then test it on the input of shape like `(batch_size, classes, dim1, ...)` for predictions/targets ?

Another point to address a bit later is to ensure that it works with DDP. In this case, we have to replicate what is done here : https://github.com/pytorch/ignite/blob/f1cc9fbb335273f4303d6a68fc7ddbb2d0610b87/tests/ignite/metrics/test_confusion_matrix.py#L783-L790

Reply:
Regarding the input format of ```(batch_size, classes, dim1, ...) ``` , is there any special use case? The tfa implementation and sklearn's function take 2d matrices.

Reply:
We can imagine, for example, image segmentation with overlapping classes...

Reply:
I see. So we can think of this as each class having ```num_batches x dim_1 x ... x dim_n``` configurations in the general case where in the 2d case, each class had ```num_batches``` number of configurations to compare between the predictions and the ground truth.

Reply:
Yes, it can be seen like that. In ConfusionMatrix implementation we however do in some sense this reshape internally and can accept inputs like `(B, C, H, W)`.

Reply:
So in the test cases we can simply compare the output of ignite's implementation with sklearn's implementation having feeding the latter with predictions and ground truth arrays after being reshaped as ```[num_batches x dim_1 x ... x dim_n , num_classes]```

Reply:
Yes, exactly ! Transposed and reshape